### ISSUE TITLE
`OP_NUM2BIN` implementation

### ISSUE THREAD

**Issue by BarbacoviF** [nchain-innovation/chain-gang#3]
The implementation of `OP_NUM2BIN` in the BSV node software pops the two top elements from the stack: `n, m`, interprets `n` as a byte string, `m` as an integer, and first minimally encodes `n`, then it extends it to `m` bytes should the minimally encoding be too short: [OP_NUM2BIN BSV software](https://github.com/bitcoin-sv/bitcoin-sv/blob/master/src/script/interpreter.cpp#L1574)

`n` is taken as a bytestring and it is not reversed, while in the current chain-gang implementation the minimal encoding of `n` is reversed before being pushed to the stack: [chain-gang OP_NUM2BIN](https://github.com/nchain-innovation/chain-gang/blob/main/src/script/interpreter.rs#L581)

The following code should return `True`, but it returns `False`

```python
from tx_engine import Script, Context

test = Script.parse_string('0x01 OP_2 OP_NUM2BIN 0x0100 OP_EQUAL')
context = Context(script=test)
context.evaluate()
```

**Comment by arthurggordon** [comment]
```
    test_script(
        # unlock=Script.parse_string('OP_1 OP_2'),
        # lock=Script.parse_string('OP_NUM2BIN 0x0100 OP_EQUAL')
    )
```
Results in {'result': 'ok'}]
```
    test_script(
        unlock=Script.parse_string('0x01 OP_2'),
        lock=Script.parse_string('OP_NUM2BIN 0x0100 OP_EQUAL')
    )
```
Results in [{'result': 'error', 'description': 'Data push larger than necessary'}]

**Comment by arthurggordon** [comment]
Addressed in version v0.6.0
